src/cache/meta_data_cache.rs

```diff
@@ -81,7 +81,12 @@ impl TryFrom<&EsedbInfo<'_>> for MetaDataCache {
         let mut record_by_guid = HashMap::new();
         let mut root = None;
         //let mut root_dse = None;
-        let count = info.data_table().count_records()?;
+        let count = match info.data_table().count_records() {
+            Ok(x) => x,
+            Err(_) => {
+                info.data_table().count_records()? as i32
+            }
+        };
```

Comment:
There seems to be no semantic difference to
```rust
let count = info.data_table().count_records()? as i32;
```
If `count_records()` failed, you call it a second time. Why should it not fail again? Is this a problem with libesedb? Do I miss something?

Reply:
Sorry, I didn't read you initial comment thoroughly enough.

Reply:
Yeah its weird. For some reason it fails the first time, but works the second time.
The error thrown:

```
Error: c-libesedb: libesedb_table_get_number_of_records: unable to retrieve number of leaf values from table page tree.
libesedb_page_tree_get_number_of_leaf_values: invalid number of leaf pages value exceeds maximum.
libesedb_table_get_number_of_records: unable to retrieve number of leaf values from table page tree.
```

Increasing `LIBESEDB_MAXIMUM_NUMBER_OF_LEAF_PAGES` does not fix the issue.

This is definitely a bug in libesedb (or maybe even a half broken `NTDS.dit`, but the active directory works fine) and not a bug in ntdsextract2, but this workaround works, so i would implement this anyways. This should be reported to libesedb, but i've honestly no idea how to even phrase it and without test data this will be difficult to fix.